### ISSUE TITLE
add support for using custom executable_opts variable via the cmd line

### ISSUE THREAD

**Issue by smoors** [EESSI/test-suite#20]
useful for testing with non-default opts

overriding the `executable` variable already works via `--setvar executable=<x>`

**Comment by smoors** [comment]
one issue is that the hpctestlib already adds options to `executable_opts`:
```
self.executable_opts += ['-nb', self.nb_impl, '-s benchmark.tpr']
```
possible solutions:
- add support for a custom variable, and if that is set via e.g. `--setvar custom_executable_opts`, and remove `-nb self.nb_impl` from `executable_opts`
- open a PR upstream to make `-nb self.nb_impl` optional. however, that would also need to support an additional custom option.
- check if any additional options are set via `--setvar executable_opts`, and remove `-nb self.nb_impl` from `executable_opts`.

the last solution is probably best, as it does not require an additional custom variable.

**Comment by smoors** [comment]
upstream PR created: https://github.com/reframe-hpc/reframe/pull/2821

**Comment by smoors** [comment]
on further thought, this seems a bad idea.
the test name always includes the self.nb_impl value, but with this change the real `-nb` value might be different, so this is misleading.

it's probably better to not add the `-nb` option on the cmd line and filter out the tests with the `nb_impl` value that you don't want.

**Comment by smoors** [comment]
reopening this
we can still support custom executable_opts, just not the `-nb` option, which is not really a problem because there are only 3 possible values: `cpu`, `gpu`, and `auto`, and `auto` means Gromacs will choose between `gpu` and `cpu`, depending on if it can find a GPU or not.

**Comment by satishskamath** [comment]
This is test specific, right and not generic? Because `cpu` and `gpu` are generic options.

**Comment by smoors** [comment]
> @smoors This is test specific, right and not generic? Because `cpu` and `gpu` are generic options.

the `-nb` option is specific for Gromacs, but it's values `cpu` and `gpu` are set by the hpctestlib, and we don't change those.

the handling of `-ntomp`, as is done in https://github.com/EESSI/test-suite/pull/23, is Gromacs-specific.

**Comment by satishskamath** [comment]
> > @smoors This is test specific, right and not generic? Because `cpu` and `gpu` are generic options.
> 
> EDIT: to be more clear: it's indeed mostly test specific, but you just made me realize that we can make the `check_custom_executable_opts` method generic, see update to #23

Yeah. I thought so too. Because any application can have this check and have custom executable options.
